app/src/main/java/com/okta/oidc/example/PlainActivity.java

```diff
@@ -448,6 +511,29 @@ public void onError(String error, AuthorizationException exception) {
         });
     }
 
+    /**
+     * Check if device have enabled keyguard.
+     *
+     * @return the boolean
+     */
+    @VisibleForTesting
+    public boolean isKeyguardSecure() {
+        KeyguardManager keyguardManager =
+                (KeyguardManager) getSystemService(Context.KEYGUARD_SERVICE);
+        return keyguardManager.isKeyguardSecure();
+    }
+
+    private void showKeyguard() {
+        KeyguardManager keyguardManager =
+                (KeyguardManager) getSystemService(Context.KEYGUARD_SERVICE);
+        Intent intent = null;
+        if (android.os.Build.VERSION.SDK_INT >= android.os.Build.VERSION_CODES.LOLLIPOP) {
+            intent = keyguardManager.createConfirmDeviceCredentialIntent("Confirm credentials", "");
```

Comment:
string resource? also this method is deprecated in api 29. If we're adding it anyways might as well use the new pattern too.

Reply:
The support library is in alpha and missing some features. For example setDeviceCredentialAllowed(boolean) is missing so we can't set an option to allow PIN or finger print. The SDKs supported API is 28 so we can't call setDeviceCredentialAllowed directly since its API 29. I have a separate PR for using biometric prompt when we officially support API 29 for the SDK. https://github.com/okta/okta-oidc-android/pull/80